### PR TITLE
Make it easier to track Structured legacy query usages

### PR DESCRIPTION
**Pull request by ranquild**
Currently it's hard to track `StructuredQuery` usage via `legacyQuery` because it could be `NativeQuery` and we don't migrate it to MLv2 now. The idea is to require `useStructuredQuery` flag in `legacyQuery` call:
```
legacyQuery({ useStructuredQuery: true })
```
otherwise it would throw an exception. If this approach gets approved I'll add `useStructuredQuery` to all places where `legacyQuery` is called, and we can migrate them one by one.

### CODE REVIEW

**Review by nemanjaglumac — COMMENTED** [review]
This approach makes a ton of sense.

I'm looking forward to seeing explicit usages of Structured query.
Thanks for doing this @ranquild

File: enterprise/frontend/src/metabase-enterprise/audit_app/components/QueryViewer/QueryViewer.tsx [hunk 1]
```diff
@@ -45,7 +42,8 @@ export function QueryViewer({ datasetQuery }: { datasetQuery: DatasetQuery }) {
     );
   }
 
-  if (query.tables()) {
+  const database = question.database();
+  if (database && database.tables) {
```
Comment (ranquild) [inline enterprise/frontend/src/metabase-enterprise/audit_app/components/QueryViewer/QueryViewer.tsx:46]: This is exactly the same check as in `query.tables()` in `StructuredQuery`. Not 100% check what this check does but I want to leave it as it is.

**Comment by uladzimirdev** [comment]
Looks nice, probably some further renaming is required, but approach LGTM

**Comment by ranquild** [comment]
@uladzimirdev I've fixed failed tests

**Review by uladzimirdev — APPROVED** [review]

**Pull request merged**